src/components/external-source/ExternalSourcesPanel.svelte

```diff
@@ -0,0 +1,161 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import {
+    derivationGroupPlanLinkError,
+    derivationGroups,
+    externalSources,
+    planDerivationGroupLinks,
+    usersSeenSources,
+  } from '../../stores/external-source';
+  import { plan } from '../../stores/plan';
+  import type { User } from '../../types/app';
+  import type { DerivationGroup, ExternalSourceSlim } from '../../types/external-source';
+  import type { ViewGridSection } from '../../types/view';
+  import effects from '../../utilities/effects';
+  import { tooltip } from '../../utilities/tooltip';
+  import Collapse from '../Collapse.svelte';
+  import CollapsibleListControls from '../CollapsibleListControls.svelte';
+  import GridMenu from '../menus/GridMenu.svelte';
+  import AlertError from '../ui/AlertError.svelte';
+  import CardList from '../ui/CardList.svelte';
+  import Panel from '../ui/Panel.svelte';
+  import ExternalSourcePanelEntry from './ExternalSourcePanelEntry.svelte';
+  import ExternalSourceUpdateCard from './ExternalSourceUpdateCard.svelte';
+
+  export let gridSection: ViewGridSection;
+  export let user: User | null;
+
+  // filter which derivation groups are visible
+  let filterText: string = '';
+  let mappedDerivationGroups: { [key: string]: DerivationGroup[] } = {};
+  let filteredDerivationGroups: DerivationGroup[] = [];
+  let unseenSources: ExternalSourceSlim[] = [];
+
+  $: console.log('userseensources', $usersSeenSources);
+
+  // Determine which new and deleted sources are unacknowledged for the user
+  $: {
+    if ($plan !== null && $plan.id !== null && $usersSeenSources[$plan.id] !== undefined) {
+      const derivationGroupsAssociated = $usersSeenSources[$plan.id];
+      const groups = Object.keys(derivationGroupsAssociated);
+      console.log(derivationGroupsAssociated);
+      unseenSources = $externalSources.filter(externalSource => {
+        if (!groups.includes(externalSource.derivation_group_name)) {
+          return false;
+        }
+        return (
+          new Date(externalSource.created_at) >
+          new Date(derivationGroupsAssociated[externalSource.derivation_group_name])
+        );
+      });
+    }
+  }
+
+  $: linkedDerivationGroupIds = $planDerivationGroupLinks
+    .filter(link => link.plan_id === $plan?.id)
+    .map(link => link.derivation_group_name);
+  $: filteredDerivationGroups = $derivationGroups
+    .filter(group => linkedDerivationGroupIds.includes(group.name))
+    .filter(group => {
+      const filterTextLowerCase = filterText.toLowerCase();
+      const includesName = group.name.toLocaleLowerCase().includes(filterTextLowerCase);
+      return includesName;
+    });
+  planDerivationGroupLinks.subscribe(_ => (mappedDerivationGroups = {})); // clear the map...
```

Comment:
The idea was to reset mapped derivation groups any time `planDerivationGroupLinks` gets an update, and then repopulate it. This reset would fire anytime `planDerivationGroupLinks` changes because it is a subscription. There wasn't a very clear way we could think about making `mappedDerivationGroups` react to `planDerivationGroupLinks`.

Reply:
I wouldn't recommend taking the flow out of the svelte lifecycle because I don't think the order of execution is predictable. The way you could do this in a svelte component is
```
$: if($planDerivationGroupLinks) {
  mappedDerivationGroups = {}
}
```

However, the way that I would recommend doing this is to change the `filteredDerivationGroups.forEach` to a `reduce` that sets `mappedDerivationGroups` as a result.
```
$: mappedDerivationGroups = filteredDerivationGroups.reduce((aggMappedDerivationGroups, group) => {
  ...
}, {});
```

Reply:
Addressed in [update ExternalSourcesPanel.svelte update logic for map](https://github.com/NASA-AMMOS/aerie-ui/commit/4a66b87e24f4f863c2f6624246e84a567cc43629).